config/ompi_setup_java.m4

```diff
@@ -13,92 +13,238 @@ dnl                         All rights reserved.
+                      AS_IF([test -n "$JAVAH"],
+                            [ompi_javah_happy=yes],
+                            [ompi_java_happy=no])])
+               rm -f Conftest.java Conftest.class Conftest.h
+
+               AS_IF([test -n "$with_jdk_headers" && test "$with_jdk_headers" != "yes" && test "$with_jdk_headers" != "no"],
+                     [OMPI_JDK_CPPFLAGS="-I$with_jdk_headers"
+                      # Some flavors of JDK also require -I<blah>/linux.
+                      # See if that's there, and if so, add a -I for that,
+                      # too.  Ugh.
+                      AS_IF([test -d "$with_jdk_headers/linux"],
+                            [OMPI_JDK_CPPFLAGS="$OMPI_JDK_CPPFLAGS -I$with_jdk_headers/linux"])
+                      # Solaris JDK also require -I<blah>/solaris.
+                      # See if that's there, and if so, add a -I for that,
+                      # too.  Ugh.
+                      AS_IF([test -d "$with_jdk_headers/solaris"],
+                            [OMPI_JDK_CPPFLAGS="$OMPI_JDK_CPPFLAGS -I$with_jdk_headers/solaris"])
+                      # Darwin JDK also require -I<blah>/darwin.
+                      # See if that's there, and if so, add a -I for that,
+                      # too.  Ugh.
+                      AS_IF([test -d "$with_jdk_headers/darwin"],
+                            [OMPI_JDK_CPPFLAGS="$OMPI_JDK_CPPFLAGS -I$with_jdk_headers/darwin"])
+
+                      CPPFLAGS="$CPPFLAGS $OMPI_JDK_CPPFLAGS"])
+               AC_CHECK_HEADER([jni.h], [],
+                               [ompi_java_happy=no])
+               CPPFLAGS=$ompi_java_CPPFLAGS_save
+              ])
     else
-        AC_MSG_RESULT([no])
-        WANT_MPI_JAVA_SUPPORT=0
+        ompi_java_happy=no
     fi
-    AC_DEFINE_UNQUOTED([OMPI_WANT_JAVA_BINDINGS], [$WANT_MPI_JAVA_SUPPORT],
-                       [do we want java mpi bindings])
-    AM_CONDITIONAL(OMPI_WANT_JAVA_BINDINGS, test "$WANT_MPI_JAVA_SUPPORT" = "1")
-
-   # Are we happy?
-    AS_IF([test "$WANT_MPI_JAVA_SUPPORT" = "1"],
-          [AC_MSG_WARN([******************************************************])
-           AC_MSG_WARN([*** Java MPI bindings are provided on a provisional])
-           AC_MSG_WARN([*** basis.  They are NOT part of the current or])
-           AC_MSG_WARN([*** proposed MPI standard.  Continued inclusion of])
-           AC_MSG_WARN([*** the Java MPI bindings in Open MPI is contingent])
-           AC_MSG_WARN([*** upon user interest and developer support.])
-           AC_MSG_WARN([******************************************************])
-          ])
-
-    AC_CONFIG_FILES([
-        ompi/mpi/java/Makefile
-        ompi/mpi/java/java/Makefile
-        ompi/mpi/java/c/Makefile
-    ])
+    AC_SUBST(OMPI_JDK_CPPFLAGS)
+
+    # Are we happy?
+    AC_MSG_CHECKING([if Java support available])
+    AS_IF([test "$ompi_java_happy" = "no"],
+          [AC_MSG_RESULT([no])],
+          [AC_MSG_RESULT([yes])])
```

Comment:
since this macro is only invoked when `--enable-mpi-java` is used, shouldn't you simply abort here ?
(or at least abort somewhere else at a higher level)